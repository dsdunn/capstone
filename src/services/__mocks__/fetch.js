export const getAllCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserCollections = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'string',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollectionsByCategory = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'}])
  })

export const getUserInfo = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666
    }])
  })

export const getSearchResults = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      collections:[]
    }])
  })

export const postCollection = jest.fn().mockImplementation(() => {
  return Promise.resolve([{  
      id: 666,
      uid: 'string',
      category: 'vinyl',
      title: 'string',
      description: 'string',
      image: 'string'
    }])
  })