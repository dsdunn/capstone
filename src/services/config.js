export default {
  apiKey: '66642069',
  url: 'https://collecshare.herokuapp.com/api/v1'
}