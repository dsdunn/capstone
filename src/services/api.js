import config from './config';

export const apiFetch = (path, method = 'GET', payload = null) => {
  let body = payload ? JSON.stringify(payload) : null
  return fetch(`${config.url}${path}`, {
    method,
    headers: {
     'Content-Type': 'application/json',
     'Accept': 'application/json'
    },
    body
  })
}

export const apiFormDataFetch = (path, method = 'PUT', payload = null) => {
  console.log(path, method, payload);
  let body = payload; 
  return fetch(`${config.url}${path}`, {
    method,
    body
  })
}
