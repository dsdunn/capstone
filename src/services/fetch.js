import { apiFetch, apiFormDataFetch } from './api'

export const postUserInfo = user => {
  return apiFetch('/users', 'POST', user)
}

export const getUserInfo = uid => {
  return apiFetch(`/users/${uid}`, 'GET')
  .then(response => response.json())
}

export const putUserInfo = (userInfo, uid) => {
  return apiFormDataFetch(`/users/${uid}`, 'PUT', userInfo)
  .then(response => console.log(response))
  .then(response => response.json())
}

export const postCollection = collection => {
  return apiFetch(`/collections`, `POST`, collection)
  .then(response => response.json())
}

export const getAllCollections = () => {
  return apiFetch('/collections')
  .then(response => response.json())
}

export const getUserCollections = uid => {
  return apiFetch(`/collections?uid=${uid}`)
  .then(response => response.json())
}