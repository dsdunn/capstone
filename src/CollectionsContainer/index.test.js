import React from 'react';
import { shallow } from 'enzyme';
import { CollectionsContainer } from './';
import { getAllCollections, getCollectionsByCategory } from '../services/fetch'

jest.mock('../services/fetch');

describe('CollectionsContainer', () => {
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    mockEvent = { preventDefault: jest.fn(),
                  target: { name: 'some cool name' } }
    wrapper = shallow(<CollectionsContainer />)

  })

  it('should get all collections when mounted', () => {
    expect(getAllCollections).toHaveBeenCalled()
  })

  it('should handle a dang click', () => {
    wrapper.instance().handleClick(mockEvent)
    const actual = wrapper.state('category')
    const expected = 'some cool name'

    expect(actual).toEqual(expected)
  })

  it('should fetch a collection by category', () => {
    wrapper.instance().fetchByCategory('vinyl')
    const actual = wrapper.state('collections')
    const expected = [{ id: 666,
             uid: 'string',
             category: 'string',
             title: 'string',
             description: 'string',
             image: 'string' }]
    expect(actual).toEqual(expected)
  })

  it('should match the snapShot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})