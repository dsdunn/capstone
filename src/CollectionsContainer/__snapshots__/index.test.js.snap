// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionsContainer should match the snapShot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CollectionsContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="collections-container-bg"
          style={
                    Object {
                              "backgroundImage": "url(collection-focus-magnifier-248993.jpg)",
                            }
          }
/>,
        <div
          className="collections-container"
>
          <h1
                    className="collections-category-name"
          >
                    all
                    .
          </h1>
          <div
                    className="collections-cat-btn-container"
          >
                    <button
                              className="collections-container-cat-btn"
                              name="comics"
                              onClick={[Function]}
                    >
                              comics
                    </button>
                    <button
                              className="collections-container-cat-btn"
                              name="cards"
                              onClick={[Function]}
                    >
                              cards
                    </button>
                    <button
                              className="collections-container-cat-btn"
                              name="coins"
                              onClick={[Function]}
                    >
                              coins
                    </button>
                    <button
                              className="collections-container-cat-btn"
                              name="vinyl"
                              onClick={[Function]}
                    >
                              vinyl
                    </button>
                    <button
                              className="collections-container-cat-btn"
                              name="other"
                              onClick={[Function]}
                    >
                              other
                    </button>
          </div>
          <withRouter(Connect(CollectionSmall))
                    collection={
                              Object {
                                        "category": "string",
                                        "description": "string",
                                        "id": 666,
                                        "image": "string",
                                        "title": "string",
                                        "uid": "string",
                                      }
                    }
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "collections-container-bg",
          "style": Object {
            "backgroundImage": "url(collection-focus-magnifier-248993.jpg)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="collections-category-name"
>
              all
              .
</h1>,
            <div
              className="collections-cat-btn-container"
>
              <button
                            className="collections-container-cat-btn"
                            name="comics"
                            onClick={[Function]}
              >
                            comics
              </button>
              <button
                            className="collections-container-cat-btn"
                            name="cards"
                            onClick={[Function]}
              >
                            cards
              </button>
              <button
                            className="collections-container-cat-btn"
                            name="coins"
                            onClick={[Function]}
              >
                            coins
              </button>
              <button
                            className="collections-container-cat-btn"
                            name="vinyl"
                            onClick={[Function]}
              >
                            vinyl
              </button>
              <button
                            className="collections-container-cat-btn"
                            name="other"
                            onClick={[Function]}
              >
                            other
              </button>
</div>,
            Array [
              <withRouter(Connect(CollectionSmall))
                collection={
                                Object {
                                                "category": "string",
                                                "description": "string",
                                                "id": 666,
                                                "image": "string",
                                                "title": "string",
                                                "uid": "string",
                                              }
                }
/>,
            ],
          ],
          "className": "collections-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "all",
                ".",
              ],
              "className": "collections-category-name",
            },
            "ref": null,
            "rendered": Array [
              "all",
              ".",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="collections-container-cat-btn"
                  name="comics"
                  onClick={[Function]}
>
                  comics
</button>,
                <button
                  className="collections-container-cat-btn"
                  name="cards"
                  onClick={[Function]}
>
                  cards
</button>,
                <button
                  className="collections-container-cat-btn"
                  name="coins"
                  onClick={[Function]}
>
                  coins
</button>,
                <button
                  className="collections-container-cat-btn"
                  name="vinyl"
                  onClick={[Function]}
>
                  vinyl
</button>,
                <button
                  className="collections-container-cat-btn"
                  name="other"
                  onClick={[Function]}
>
                  other
</button>,
              ],
              "className": "collections-cat-btn-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "comics",
                  "className": "collections-container-cat-btn",
                  "name": "comics",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "comics",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "cards",
                  "className": "collections-container-cat-btn",
                  "name": "cards",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "cards",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "coins",
                  "className": "collections-container-cat-btn",
                  "name": "coins",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "coins",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "vinyl",
                  "className": "collections-container-cat-btn",
                  "name": "vinyl",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "vinyl",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "other",
                  "className": "collections-container-cat-btn",
                  "name": "other",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "other",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "collection": Object {
                "category": "string",
                "description": "string",
                "id": 666,
                "image": "string",
                "title": "string",
                "uid": "string",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="collections-container-bg"
            style={
                        Object {
                                    "backgroundImage": "url(collection-focus-magnifier-248993.jpg)",
                                  }
            }
/>,
          <div
            className="collections-container"
>
            <h1
                        className="collections-category-name"
            >
                        all
                        .
            </h1>
            <div
                        className="collections-cat-btn-container"
            >
                        <button
                                    className="collections-container-cat-btn"
                                    name="comics"
                                    onClick={[Function]}
                        >
                                    comics
                        </button>
                        <button
                                    className="collections-container-cat-btn"
                                    name="cards"
                                    onClick={[Function]}
                        >
                                    cards
                        </button>
                        <button
                                    className="collections-container-cat-btn"
                                    name="coins"
                                    onClick={[Function]}
                        >
                                    coins
                        </button>
                        <button
                                    className="collections-container-cat-btn"
                                    name="vinyl"
                                    onClick={[Function]}
                        >
                                    vinyl
                        </button>
                        <button
                                    className="collections-container-cat-btn"
                                    name="other"
                                    onClick={[Function]}
                        >
                                    other
                        </button>
            </div>
            <withRouter(Connect(CollectionSmall))
                        collection={
                                    Object {
                                                "category": "string",
                                                "description": "string",
                                                "id": 666,
                                                "image": "string",
                                                "title": "string",
                                                "uid": "string",
                                              }
                        }
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "collections-container-bg",
            "style": Object {
              "backgroundImage": "url(collection-focus-magnifier-248993.jpg)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="collections-category-name"
>
                all
                .
</h1>,
              <div
                className="collections-cat-btn-container"
>
                <button
                                className="collections-container-cat-btn"
                                name="comics"
                                onClick={[Function]}
                >
                                comics
                </button>
                <button
                                className="collections-container-cat-btn"
                                name="cards"
                                onClick={[Function]}
                >
                                cards
                </button>
                <button
                                className="collections-container-cat-btn"
                                name="coins"
                                onClick={[Function]}
                >
                                coins
                </button>
                <button
                                className="collections-container-cat-btn"
                                name="vinyl"
                                onClick={[Function]}
                >
                                vinyl
                </button>
                <button
                                className="collections-container-cat-btn"
                                name="other"
                                onClick={[Function]}
                >
                                other
                </button>
</div>,
              Array [
                <withRouter(Connect(CollectionSmall))
                  collection={
                                    Object {
                                                      "category": "string",
                                                      "description": "string",
                                                      "id": 666,
                                                      "image": "string",
                                                      "title": "string",
                                                      "uid": "string",
                                                    }
                  }
/>,
              ],
            ],
            "className": "collections-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "all",
                  ".",
                ],
                "className": "collections-category-name",
              },
              "ref": null,
              "rendered": Array [
                "all",
                ".",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <button
                    className="collections-container-cat-btn"
                    name="comics"
                    onClick={[Function]}
>
                    comics
</button>,
                  <button
                    className="collections-container-cat-btn"
                    name="cards"
                    onClick={[Function]}
>
                    cards
</button>,
                  <button
                    className="collections-container-cat-btn"
                    name="coins"
                    onClick={[Function]}
>
                    coins
</button>,
                  <button
                    className="collections-container-cat-btn"
                    name="vinyl"
                    onClick={[Function]}
>
                    vinyl
</button>,
                  <button
                    className="collections-container-cat-btn"
                    name="other"
                    onClick={[Function]}
>
                    other
</button>,
                ],
                "className": "collections-cat-btn-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "comics",
                    "className": "collections-container-cat-btn",
                    "name": "comics",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "comics",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "cards",
                    "className": "collections-container-cat-btn",
                    "name": "cards",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "cards",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "coins",
                    "className": "collections-container-cat-btn",
                    "name": "coins",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "coins",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "vinyl",
                    "className": "collections-container-cat-btn",
                    "name": "vinyl",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "vinyl",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "other",
                    "className": "collections-container-cat-btn",
                    "name": "other",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "other",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "collection": Object {
                  "category": "string",
                  "description": "string",
                  "id": 666,
                  "image": "string",
                  "title": "string",
                  "uid": "string",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
